### PR TITLE
Bazaar pallet

### PR DESCRIPTION
**Pull request by haerdib**
Bazaar pallet providing shop admission possibility. Articles not yet fully implemented.

### CODE REVIEW

File: bazaar/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+[dependencies.encointer-currencies]
+default-features = false
+path = "../currencies"
+package = "pallet-encointer-currencies"
+
+[dependencies.runtime-io]
+default-features = false
+package = "sp-io"
+version = "2.0.0"
```
Comment (brenzi) [inline bazaar/Cargo.toml:28]: master has progressed to 

```
git = "https://github.com/paritytech/substrate.git" 
rev = "a208da16"
```
see other pallets.

please update your dependencies too

File: bazaar/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+use encointer_currencies::{CurrencyIdentifier};
+
+pub trait Trait: frame_system::Trait 
+    + encointer_currencies::Trait 
+{
+    type Event: From<Event<Self>> + Into<<Self as frame_system::Trait>::Event>;
+}
+
+pub type ShopIdentifier = u64;
```
Comment (brenzi) [inline bazaar/src/lib.rs:51]: I doubt this is the right branch. Shouldn't ShopIdentifier be an IPFS hash?

**Review by brenzi — CHANGES_REQUESTED** [review]
I suspect you actually wanted to merge the add-article branch, not this one.
Please rebase that branch on master as we had previously merged parts of it

File: bazaar/src/lib.rs (hunk 2 above)
Comment (haerdib) [reply bazaar/src/lib.rs:51]: Sorry, you're right. I will update the dependencies and make a new PR.

**Pull request closed without merging**